From 9fef211a5fcd2a69b666023350c4ddcbca5e085f Mon Sep 17 00:00:00 2001
From: Stefan Agner <stefan.agner@toradex.com>
Date: Tue, 11 Aug 2020 10:37:28 +0200
Subject: [PATCH] Allow custom Debian package dependencies

Allow to pass custom Debian package dependencies. This allows to use a
different JRE than hardcoded in the CMakeLists.txt, e.g.
openjdk-11-jre-headless to build for Debian Buster.

Signed-off-by: Stefan Agner <stefan.agner@toradex.com>
---
 CMakeLists.txt | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 3e4d3a075..a75dd2718 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -328,6 +328,10 @@ if(BUILD_SOTA_TOOLS)
     set(CPACK_COMPONENT_GARAGE_DEPLOY_DESCRIPTION "garage-deploy utility")
     set(CPACK_DEBIAN_GARAGE_DEPLOY_PACKAGE_SHLIBDEPS ON)
     set(CPACK_DEBIAN_GARAGE_DEPLOY_PACKAGE_DEPENDS "openjdk-8-jre")
+    if(SOTA_DEBIAN_PACKAGE_DEPENDS)
+      set(CPACK_DEBIAN_GARAGE_DEPLOY_PACKAGE_DEPENDS ${SOTA_DEBIAN_PACKAGE_DEPENDS})
+    endif()
+    message(STATUS "Using ${CPACK_DEBIAN_GARAGE_DEPLOY_PACKAGE_DEPENDS} as Debian package depends.")
 endif(BUILD_SOTA_TOOLS)
 
 set(CPACK_DEBIAN_AKTUALIZR_PACKAGE_NAME "aktualizr")
-- 
2.27.0

